src/library-authoring/author-library/BlockPreview.jsx

```diff
@@ -0,0 +1,277 @@
+  getXBlockHandlerUrl,
+  libraryBlockShape,
+  libraryShape,
+  LOADING_STATUS,
+  ROUTES,
+  XBLOCK_VIEW_SYSTEM,
+  fetchable,
+} from '../common';
+import { LoadingPage } from '../../generic';
+import messages from './messages';
+import {
+  deleteLibraryBlock,
+  fetchLibraryBlockMetadata,
+  fetchLibraryBlockView,
+  initializeBlock,
+} from '../edit-block/data';
+import { blockStatesShape, blockViewShape } from '../edit-block/data/shapes';
+import commonMessages from '../common/messages';
+import selectLibraryDetail from '../common/data/selectors';
+
+ensureConfig(['STUDIO_BASE_URL'], 'library API service');
+const getHandlerUrl = async (blockId) => getXBlockHandlerUrl(blockId, XBLOCK_VIEW_SYSTEM.Studio, 'handler_name');
+/**
+ * BlockPreviewBase
+ * Template component for BlockPreview cards, which are used to display
+ * components and render controls for them in a library listing.
+ */
+const BlockPreviewBase = ({
+  intl, block, view, canEdit, showPreviews, showDeleteModal,
+  setShowDeleteModal, library, previewKey, editView, isLtiUrlGenerating,
+  ...props
+}) => (
+  <>
+    <Card className="block-preview">
+      <Card.Header
+        title={block.display_name}
+        className={showPreviews && 'bottom-not-rounded'}
+        actions={(
+          <ActionRow>
+            { library.allow_lti && (
+              <Button
+                disabled={isLtiUrlGenerating}
+                size="sm"
+                className="mr-1"
+                onClick={() => { props.fetchBlockLtiUrl({ blockId: block.id }); }}
+              >
+                <FontAwesomeIcon icon={faClipboard} className="pr-1" />
+                {intl.formatMessage(messages['library.detail.block.copy_lti_url'])}
+              </Button>
+            )}
+            <Link to={editView}>
+              <Button size="sm">
+                <FontAwesomeIcon icon={faEdit} className="pr-1" />
+                {intl.formatMessage(messages['library.detail.block.edit'])}
+              </Button>
+            </Link>
+            <IconButton
+              isActive
+              aria-label={intl.formatMessage(messages['library.detail.block.delete'])}
+              src={Delete}
+              iconAs={Icon}
+              alt={intl.formatMessage(messages['library.detail.block.delete'])}
+              onClick={() => setShowDeleteModal(true)}
+              className="bg-light-200"
+              invertColors
+            />
+          </ActionRow>
+        )}
+      />
+      {showPreviews && (
+        <>
+          <Card.Divider />
+          <Card.Body>
+            <LibraryBlock
+              getHandlerUrl={getHandlerUrl}
+              view={view}
+              key={previewKey}
+            />
+          </Card.Body>
+        </>
+      )}
+    </Card>
+    <AlertModal
+      isOpen={showDeleteModal}
+      title={intl.formatMessage(messages['library.detail.block.delete.modal.title'])}
+      footerNode={(
+        <ActionRow>
+          <Button
```

Comment:
Do we need to add the `size` prop here and to the next button?

Reply:
I don't think it's necessary because we don't have a design
previous modal with unsaved changes has `md` size

Reply:
I don't think it's necessary because by default has `md` size
previous modal with unsaved changes has md size
But for consistency I will add